italanta/elewa
# ✨ Mollie Payments Integration Backend

# Description

As an organisation, it would be good to have an automated system where I pay for CLM services depending on my needs. 

I have integrated Mollie payments for a one-off purchase and subscriptions. 

### Details on how it works:

You first create a payment, and request a customer to pay by sending them a checkout link. The checkout links directs them to Mollie's payment page and they can go ahead and complete a payment. 
Goomza will be notified of the payment's status through a webhook POST by Mollie, containing a transaction detail. 

For subscriptions, we first create a Mollie customer, whose I.d. will be linked to subsequent changes and details to their subscription. 
They then make a first payment, and allow mollie to do subsequent payments through a mandate. 
Upon successful payments, we record the transactions in firebase under the user/activeOrg/transactions

See Mollie documentations here [https://docs.mollie.com/reference/v2/payments-api/create-payment](url)

For this project, you will need to install Mollie API Client 
npm i @mollie/api-client --legacy-peer-deps

## Type of change

Please delete options that are not relevant.


# Screenshot (optional)

# How Has This Been Tested?
Using Postman, 
- Create a customer using the creatMollieUser function url,
- use the customer to create a subscription,
- check the webhook for transactions updates,
- successful transactions are added to a user's profile under the  transaction document


**Test Configuration**: (optional)
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:
